src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheck.java

```diff
@@ -317,6 +338,45 @@ public void visitToken(DetailAST ast) {
         }
     }
 
+    /**
+     * Process lambda parameters.
+     *
+     * @param lambdaAst node of type {@link TokenTypes#LAMBDA}
+     */
+    private void processLambdaParameters(DetailAST lambdaAst) {
+        final DetailAST lambdaParameters = lambdaAst.findFirstToken(TokenTypes.PARAMETERS);
+        if (lambdaParameters != null) {
+            TokenUtil.forEachChild(lambdaParameters, TokenTypes.PARAMETER_DEF,
+                    this::checkUnnamedVariables);
+        }
+    }
+
+    /**
+     * Check if the variable is unnamed and has redundant final modifier.
+     *
+     * @param ast node of type {@link TokenTypes#VARIABLE_DEF}
+     *     or {@link TokenTypes#PATTERN_VARIABLE_DEF}
+     *     or {@link TokenTypes#PARAMETER_DEF}
+     */
+    private void checkUnnamedVariables(DetailAST ast) {
+        if (jdkVersion >= JDK_22 && isUnnamedVariable(ast)) {
+            checkForRedundantModifier(ast, TokenTypes.FINAL);
+        }
+    }
+
+    /**
+     * Check if the variable is unnamed.
+     *
+     * @param ast node of type {@link TokenTypes#VARIABLE_DEF}
+     *     or {@link TokenTypes#PATTERN_VARIABLE_DEF}
+     *     or {@link TokenTypes#PARAMETER_DEF}
+     * @return true if the variable is unnamed
+     */
+    private static boolean isUnnamedVariable(DetailAST ast) {
+        return "_".equals(ast.findFirstToken(TokenTypes.IDENT)
+                .getText());
```

Comment:
`        return "_".equals(ast.findFirstToken(TokenTypes.IDENT).getText());`


may be in future we can make check that hint that wrapped code can be single line.

Reply:
alternative formatting is:
```
"_".equals(
    ast.findFirstToken(TokenTypes.IDENT).getText());
```

Reply:
@romani done